### PR TITLE
feat: add support for tables

### PR DESCRIPTION
**Pull request by jhchabran**
As I was going through what's left to import the dev docs, noticed we had a few tables over there, which this PR provides support for. 

I also fixed a nasty bug when inserting top level elements, which was due to incorrectly handling the cursor when inserting headings. 

Finally, there's a curious hardcoded `200` value (with a comment of course), which caused me to pull my hairs for a while: when you pass the `extensions.GFM` for table support, the underlying code is also attaching rendering functions for HTML rendering them, with a priority of `500`. 

While this is quite suprising, it's in the `goldmark` code and the rationale for it is I suppose that it's just more convenient. I think we could untangle that by using the lower level API for the configuring options, but it's not very much worth the effort at this stage.

### CODE REVIEW

**Comment by bobheadxi** [comment]
> I also fixed a nasty bug when inserting top level elements, which was due to incorrectly handling the cursor when inserting headings.

This doesn't seem reflected in any test cases, should we add one to cover this?

File: testdata/ref/table.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# We can use tables too
+
+Glorious days ahead! 
+
+| A | B | C |
+|---|---|---| 
+| 1 | 1 | 1 | 
+| *foo* is a common _term_ | _bar_ | **baz** |
+| [link](https://github.com) | `code` | |
```
Comment (bobheadxi) [inline testdata/ref/table.md:9]: Should we add the various table column variants as well? e.g. centering and whatnot

**Review by bobheadxi — APPROVED** [review]
would be cool to have some integration testing as well, over the same test dataset in `testdata`, to preview how some of these changes look live 😁

File: renderer/cursor.go [hunk 2]
```diff
@@ -72,6 +74,10 @@ func (c *cursor) AppendRichText(rt *notionapi.RichText) {
 		block.Quote.RichText = append(block.Quote.RichText, rts...)
 	case *notionapi.CalloutBlock:
 		block.Callout.RichText = append(block.Callout.RichText, rts...)
+	case *notionapi.TableBlock:
+		panic(rts[0].Text.Content)
```
[the remark this answers is not in the document]
Comment (jhchabran) [reply renderer/cursor.go:78]: left over 😅

**Comment by jhchabran** [comment]
> > I also fixed a nasty bug when inserting top level elements, which was due to incorrectly handling the cursor when inserting headings.
> 
> This doesn't seem reflected in any test cases, should we add one to cover this?

It's covered by the table golden tests, the bug was breaking them to begin with. So it is tested :D

File: testdata/ref/table.md (hunk 1 above)
Comment (bobheadxi) [reply testdata/ref/table.md:9]: I think adding the column variants here is just to:

1. Demonstrate the rendering doesn't implode, even if we don't implement it correctly
2. Demonstrate the diff between "we didn't implement it" and "we've implemented" better at review time, when we get around to adding it

WDYT?

File: notionreposync.go [hunk 3]
```diff
@@ -22,12 +22,9 @@ func Import(ctx context.Context, client *notionapi.Client, repo *Repo, nd *Notio
 	}
 
 	err := repo.Walk(func(d *Document) error {
-		// if d.Path != "index.md" && d.Path != "ref/ol.md" {
-		// 	return nil
-		// }
 		println("🦀", "rendering", d.Path)
```
Comment (burmudar) [inline notionreposync.go:25]: debugging?

Comment (jhchabran) [reply notionreposync.go:25]: Yes

**Review by burmudar — APPROVED** [review]

**Pull request merged**